test/metabase/search/filter_test.clj

```diff
@@ -72,15 +85,67 @@
                     (merge default-search-ctx
                            {:created-by 1}))))))
 
-  (testing "throw error for filtering with unsupport models"
-    (is (thrown-with-msg?
-         clojure.lang.ExceptionInfo
-         #":created-by filter for database is not supported"
-         (search.filter/build-filters
-          base-search-query
-          "database"
-          (merge default-search-ctx
-                 {:created-by 1}))))))
+  (testing "verified filter"
```

Comment:
Why do we need unit tests of `build-filters` when we have the tests at the API layer in metabase.api.search-test? I think it's preferable to cover all corner cases with api tests anyway.

In any case, You are using `=?` for these tests, which is dangerous. There could be extra keys in the clause that have meaningful impact on the search results, such as joins. If they were present the tests wouldn't catch them

Reply:
> Why do we need unit tests of build-filters when we have the tests at the API layer in metabase.api.search-test?

I think the tests in  API cover all cases already. but the code for search has too many layers of logic that I think it's good to have an intermediate layer of test.

it's good to see how the query looks like too I think, rather than just checking the behavior.

> You are using =? for these tests, which is dangerous.

agree, I'll switch to the good old `=`.